tls/s2n_security_policies.c

```diff
@@ -1135,7 +1144,7 @@ bool s2n_pq_kem_is_extension_required(const struct s2n_security_policy *security
         return false;
     }
     for (uint8_t i = 0; i < cipher_preferences->count; i++) {
-        if (s2n_cipher_suite_requires_pq_extension(cipher_preferences->suites[i])) {
+        if (s2n_cipher_suite_requires_pq_extension(cipher_preferences->suites[i], security_policy->kem_preferences)) {
```

Comment:
I don't think the kem_preferences check should be part of `s2n_**cipher_suite**_requires_pq_extension`.

Instead, you could:
* Make a new `s2n_security_policy_requires_pq_extension` method that checks both cipher_preferences and kem_preferences (since it looks like we always call s2n_cipher_suite_requires_pq_extension in a loop over all cipher_preferences anyway)
* Just make a separate `if (security_policy->kem_preferences->kem_count == 0)` check in both this method and `s2n_security_policies_init`. You certainly don't need to make that check in a loop in either case.

Reply:
sure, that makes sense. i'll remove this parameter from `s2n_cipher_suite_requires_pq_extension`.